numba_cuda/numba/cuda/cudadrv/nvrtc.py

```diff
@@ -251,6 +254,31 @@ def compile(src, name, cc, ltoir=False):
     nvrtc = NVRTC()
     program = nvrtc.create_program(src, name)
 
+    version = nvrtc.get_version()
+    if version < (11, 0):
+        # Maximum supported target architecture for 11.x is cc 9.0 in CUDA 11.8
+        raise RuntimeError(
+            "Unsupported CUDA version. CUDA 11.0 or higher is required."
+        )
+    else:
+        min_cc, max_cc = CTK_SUPPORTED[version]
+        if version < min_cc:
+            raise RuntimeError(
+                f"Device Compute Capability ({cc}) is lower "
+                f"than minimum supported CC ({min_cc}) with installed "
+                "cudatoolkit. Consider downgrading cudatoolkit to supported "
+                "version."
+            )
```

Comment:
Please replace `cudatoolkit` with something more specific that shows the NVRTC version that was detected. Also, I don't think it is very helpful to recommend downgrading CUDA here. It isn't very actionable without more information, and I don't think it should be numba-cuda's job to recommend which NVRTC/CUDA Toolkit/driver versions support which devices.
```suggestion
            raise RuntimeError(
                f"Device Compute Capability ({cc}) is lower than the "
                f"minimum supported Compute Capability ({min_cc}) with "
                f"NVRTC {".".join(version)}."
            )
```